src/browser/api/application.js

```diff
@@ -1067,4 +1102,97 @@ function isNonEmptyString(str) {
     return typeof str === 'string' && str.length > 0;
 }
 
+// duck-type a possible array of Preload objects
+function isPreloads(preloads) {
+    return Array.isArray(preloads) && preloads.every(isPreload);
+}
+
+// duck-type a possible Preload object
+function isPreload(preload) {
+    return (
+        typeof preload === 'object' &&
+        typeof preload.url === 'string'
+    );
+}
+
+/**
+ * Asynchronously fetches applications's preload script(s) _to_ cache and reads in _from_ cache.
+ * ```typescript
+ * interface Preload { url:string; optional?:boolean; name?:string; };
+ * interface DecoratedPreload extends Preloader { script:string; description:string };
+ * ```
+ * @param identity {{uuid:string}}
+ * @param preloads {Preload[]}
+ * @param callback {function(Error|DecoratedPreload[])} - Called with one of:
+ * * `Error` - on first required fetch or read failure; or
+ * * `DecoratedPreload[]` - on success of all loaded scripts.
+ */
+function getPreloadScript(identity, preloads, callback) {
```

Comment:
As agreed in conversation, will create a ticket todo this later. (Actually I already coded it last night but let's not hold this up any longer.)